hack/build

```diff
@@ -15,70 +15,101 @@ CLI_BINARY_NAME=pouch
 
 function pre()
 {
-    mkdir -p $BUILDPATH
+	mkdir -p $BUILDPATH
 
-    mkdir -p $BUILDPATH/src/github.com/docker
-    [ -L $BUILDPATH/src/github.com/docker/libnetwork ] && rm -f $BUILDPATH/src/github.com/docker/libnetwork
-    ln -s $DIR/extra/libnetwork $BUILDPATH/src/github.com/docker/libnetwork
+	mkdir -p $BUILDPATH/src/github.com/docker
+	[ -L $BUILDPATH/src/github.com/docker/libnetwork ] && rm -f $BUILDPATH/src/github.com/docker/libnetwork
+	ln -s $DIR/extra/libnetwork $BUILDPATH/src/github.com/docker/libnetwork
 
-    mkdir -p $BUILDPATH/src/github.com/alibaba
-    [ -L $BUILDPATH/src/github.com/alibaba/pouch ] && rm -f $BUILDPATH/src/github.com/alibaba/pouch
-    ln -s $DIR $BUILDPATH/src/github.com/alibaba/pouch
+	mkdir -p $BUILDPATH/src/github.com/alibaba
+	[ -L $BUILDPATH/src/github.com/alibaba/pouch ] && rm -f $BUILDPATH/src/github.com/alibaba/pouch
+	ln -s $DIR $BUILDPATH/src/github.com/alibaba/pouch
 }
 
 function server()
 {
-    cd $BUILDPATH/src/github.com/alibaba/pouch
-    echo "GOOS=linux $GOBUILD -o $BINARY_NAME"
-    GOOS=linux $GOBUILD -o $BINARY_NAME -tags 'selinux'
+	cd $BUILDPATH/src/github.com/alibaba/pouch
+	echo "GOOS=linux $GOBUILD -o $BINARY_NAME"
+	GOOS=linux $GOBUILD -o $BINARY_NAME -tags 'selinux'
 }
 
 function client()
 {
-    cd $BUILDPATH/src/github.com/alibaba/pouch
-    echo "$GOBUILD -o $CLI_BINARY_NAME github.com/alibaba/pouch/cli"
-    $GOBUILD -o $CLI_BINARY_NAME github.com/alibaba/pouch/cli
+	cd $BUILDPATH/src/github.com/alibaba/pouch
+	echo "$GOBUILD -o $CLI_BINARY_NAME github.com/alibaba/pouch/cli"
+	$GOBUILD -o $CLI_BINARY_NAME github.com/alibaba/pouch/cli
+}
+
+function testserver()
+{
+set -x
+	cd $BUILDPATH/src/github.com/alibaba/pouch
+	#
+	# Do not calculate the coverage for the following pkgs
+	#
+	skip_pkgs=(\
+		"github.com/alibaba/pouch/vendor" \
+		"github.com/alibaba/pouch/cli" \
+		"github.com/alibaba/pouch/cli/inspect" \
+		"github.com/alibaba/pouch/credential" \
+		"github.com/alibaba/pouch/extra" \
+		"github.com/alibaba/pouch/pkg/kmutex" \
+		"github.com/alibaba/pouch/pkg/term" \
+		"github.com/alibaba/pouch/pkg/utils/templates" \
+		"github.com/alibaba/pouch/plugins" \
+		"github.com/alibaba/pouch/storage/volume/examples/demo" \
+		"github.com/alibaba/pouch/test" \
+	)
+	cmd="go list ./... "
+	for j in $(seq 0 $(( ${#skip_pkgs[@]} -1)) ); do
+		cmd+="| grep -v \"${skip_pkgs[$j]}\" "
+	done
+	echo $cmd
+
+	pkgs=$(eval $cmd |tr '\n' ','|sed 's/,$//g')
+
+	go test -c -race -cover -covermode=atomic -o pouchd-test -coverpkg $pkgs
 }
 
 function vet()
 {
-    cd $BUILDPATH/src/github.com/alibaba/pouch
-    GOPACKAGES=`go list ./... | grep -v /vendor/ | grep -v /extra/ | sed 's/^_//'`
-    go vet $GOPACKAGES 2>&1 | grep -v "unrecognized printf verb 'r'" | egrep -v "exit status 1" | tee /dev/stderr
+	cd $BUILDPATH/src/github.com/alibaba/pouch
```

Comment:
Please use unified 4 four whitespaces instead of tab. I do not think mixing both is a gooad way.